apps/studio/prisma/schema.prisma

```diff
@@ -26,68 +26,86 @@ model VerificationToken {
-  deletedAt DateTime? @db.Timestamp
-
-  @@index([parentPostId])
-  @@index([parentPostId, deletedAt, id])
-  @@index([deletedAt])
-  @@index([createdAt])
+model Resource {
+  id             String       @id @default(cuid())
+  name           String
+  siteId         String
+  parentId       String?
+  lastEditTime   DateTime?
+  lastEditUserId String
+  blobId         String?      @unique
+  status         PageStatus // Is this a good place to be storing this?
+  site           Site         @relation(fields: [siteId], references: [id])
+  parent         Resource?    @relation("parentRelation", fields: [parentId], references: [id])
+  children       Resource[]   @relation("parentRelation")
+  lastEditUser   User?        @relation(fields: [lastEditUserId], references: [id], onDelete: NoAction, onUpdate: NoAction) // Don't delete the file if the user is deleted
+  blob           Blob?
+  permission     Permission[]
+
+  @@unique([siteId, parentId, name], name: "uniqueChildNames")
 }
 
-model LikedPosts {
-  post      Post     @relation(fields: [postId], references: [id], onDelete: NoAction, onUpdate: NoAction)
-  postId    String
-  user      User     @relation(fields: [userId], references: [id], onDelete: NoAction, onUpdate: NoAction)
-  userId    String
-  createdAt DateTime @default(now())
+model User {
+  id    String @id @default(cuid())
+  name  String
+  email String @unique
+  phone String
 
-  @@id([postId, userId])
+  Resource     Resource[]
+  Permission   Permission[]
+  SitesOnUsers SitesOnUsers[]
 }
 
-model User {
-  id            String       @id @default(cuid())
-  name          String?
-  username      String?      @unique
-  email         String?      @unique
-  emailVerified DateTime?
-  image         String?
-  bio           String?
-  posts         Post[]
-  likedPosts    LikedPosts[]
-  accounts      Accounts[]
+model Permission {
+  id         String   @id @default(cuid())
+  resourceId String
+  userId     String
+  role       RoleType
+
+  // Delete the row if either resource or user is deleted, default behavior cascade
+  resource Resource @relation(fields: [resourceId], references: [id])
+  user     User     @relation(fields: [userId], references: [id])
+
+  // userId and resource Id should be unique? Each user should have unambiguous access to resource? Or not if we take the strictest or laxest perm for the user
+  @@unique([userId, resourceId], name: "unambiguousPermissionsPerUserResource")
+}
+
+model Site {
+  id           String         @id @default(cuid())
+  name         String         @unique
+  Resource     Resource[]
+  SitesOnUsers SitesOnUsers[]
 }
 
-model Accounts {
-  id                String @id @default(cuid())
-  provider          String
-  providerAccountId String
+model Blob {
+  id      String @id @default(cuid())
+  content Json
 
+  // Should the blob be deleted on deletion of the resource? 
+  resource Resource @relation(fields: [id], references: [blobId])
```

Comment:
we should make the be on resource because most (if not all) of the time, the query will be from resource -> blob.

Reply:
Should be rectified!